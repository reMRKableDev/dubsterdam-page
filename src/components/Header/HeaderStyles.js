import styled from "styled-components";
import { bgColor, brandColor } from "../../utils/themeColors";
import { respondTo } from "../../layout/respondTo";
import { fadeInUpAnimation, shakeAnimation } from "../../utils/animations";

export const StyledHeader = styled.header`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
`;

export const StyledFigure = styled.figure`
  animation: 1.2s ${shakeAnimation};
  animation-delay: 1.4s;
  height: 100%;
  width: 100%;
  margin: 0;
  padding: 0;

  img {
    width: 30vw;
    height: auto;
  }
`;

export const StyledH1 = styled.h1`
  animation: 1.5s ${fadeInUpAnimation};
  font-size: 8vw;
  font-family: "Rubik", sans-serif;
  margin: 1em 0;
  text-transform: uppercase;

  span {
    color: ${brandColor};
    border: 4px solid ${brandColor};
    box-shadow: 0 0 10px ${!bgColor};
    padding: 5px;
    text-transform: none;
  }

  ${respondTo.sm`
    margin: 0.8em 0;

    span {
      border: 6px solid ${brandColor};
    }
  `}

  ${respondTo.md`
  margin: 0.6em 0;

  span {
    border: 8px solid ${brandColor};
  }
  `}
  
  ${respondTo.lg`
  margin: 0.5em 0;

  span {
    border: 10px solid ${brandColor};
  }
	`}
`;
