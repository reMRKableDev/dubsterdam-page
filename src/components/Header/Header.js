import React from "react";
import wcsLogo from "../../assets/logo.png";

import { StyledHeader, StyledFigure, StyledH1 } from "./HeaderStyles";

const Header = () => {
  return (
    <StyledHeader>
      <StyledFigure>
        <img src={wcsLogo} alt="logo" />
      </StyledFigure>
      <StyledH1>
        We are <span>DubSterdam</span>
      </StyledH1>
    </StyledHeader>
  );
};

export default Header;
