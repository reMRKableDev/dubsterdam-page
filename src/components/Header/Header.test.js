import React from "react";
import { shallow, render } from "enzyme";
import Header from "./Header";
import wcsLogo from "../../assets/logo.png";

import {
  validateMatchSnapshot,
  validateToEqual,
} from "../../utils/testValidators";

describe("Header unit tests", () => {
  it("should validate snapshot of Header component", () => {
    const wrapper = shallow(<Header />);
    validateMatchSnapshot(wrapper);
    validateToEqual(wrapper.find("img").prop("src"), wcsLogo);
  });

  it("should validate <img> src", () => {
    const wrapper = shallow(<Header />);
    const src = wrapper.find("img").prop("src");
    validateToEqual(src, wcsLogo);
  });

  it("should validate <img> src", () => {
    const wrapper = render(<Header />);
    const h1Text = wrapper.find("h1").text();
    validateToEqual(h1Text, "We are DubSterdam");
  });
});
