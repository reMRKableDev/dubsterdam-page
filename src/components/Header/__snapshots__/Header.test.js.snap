// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header unit tests should validate snapshot of Header component 1`] = `
<styled.header>
  <styled.figure>
    <img
      alt="logo"
      src="logo.png"
    />
  </styled.figure>
  <styled.h1>
    We are 
    <span>
      DubSterdam
    </span>
  </styled.h1>
</styled.header>
`;
